File: rb/features.rb
# CarbonIntensity SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module CarbonIntensityFeatures
  def self.make_feature(name)
    case name
    when "base"
      CarbonIntensityBaseFeature.new
    when "ratelimit"
      CarbonIntensityRatelimitFeature.new
    when "retry"
      CarbonIntensityRetryFeature.new
    when "test"
      CarbonIntensityTestFeature.new
    when "timeout"
      CarbonIntensityTimeoutFeature.new
    else
      CarbonIntensityBaseFeature.new
    end
  end
end

File: py/features.py
# CarbonIntensity SDK feature factory

from feature.base_feature import CarbonIntensityBaseFeature
from feature.test_feature import CarbonIntensityTestFeature


def _make_feature(name):
    features = {
        "base": lambda: CarbonIntensityBaseFeature(),
        "test": lambda: CarbonIntensityTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
